src/api/walker.ts

```diff
@@ -88,13 +90,16 @@ export class Walker<TOutput extends Output> {
         useRealPaths,
         pathSeparator,
       },
+      controller,
     } = this.state;
 
-    if ((signal && signal.aborted) || (maxFiles && paths.length > maxFiles))
+    if (
+      controller.signal.aborted ||
+      (signal && signal.aborted) ||
+      (maxFiles && paths.length > maxFiles)
+    )
       return;
 
-    this.pushDirectory(directoryPath, paths, filters);
```

Comment:
does this change mean the root directory isn't pushed anymore? presumably `directoryPath` can be `/`, so this change would mean we push the child directories but not `/` itself.

that may be right, just checking i understand the logic if anything